classes/order/Order.php

```diff
@@ -952,11 +952,11 @@ public static function getCustomerOrders($id_customer, $show_hidden_status = fal
         }
 
         foreach ($res as $key => $val) {
-            // In case order creation crashed midway some data might be absent
+            // In case state is hidden or order creation crashed midway some data might be absent
             $orderState = !empty($val['id_order_state']) ? $indexedOrderStates[$val['id_order_state']] : null;
-            $res[$key]['order_state'] = $orderState['name'] ?: null;
-            $res[$key]['invoice'] = $orderState['invoice'] ?: null;
-            $res[$key]['order_state_color'] = $orderState['color'] ?: null;
+            $res[$key]['order_state'] = $orderState ? $orderState['name'] : null;
+            $res[$key]['invoice'] = $orderState ? $orderState['invoice'] : null;
+            $res[$key]['order_state_color'] = $orderState ? $orderState['color'] : null;
```

Comment:
```suggestion
            $res[$key]['order_state_color'] = $orderState ? ($orderState['color'] ?: null) : null;
            $res[$key]['invoice'] = $orderState ? ($orderState['invoice'] ?: null) : null;
            $res[$key]['order_state_color'] = $orderState ? ($orderState['color']  ?: null) : null;
```

Reply:
Thanks for your review @Progi1984 ! You're right I should better check that `$orderState` array.

I have to use `!empty()` or `isset()` instead of your suggestion though. Code you suggested may still result in `Notice`, please consider following:
```
$orderState = [ ];
var_dump($orderState['color'] ?: null);
```
```
PHP Notice:  Undefined index: color
```